domains/dfinity/validation.go

```diff
@@ -0,0 +1,72 @@
+package dfinity
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"hash/crc32"
+	"strings"
+
+	"github.com/MixinNetwork/mixin/crypto"
+)
+
+const (
+	DfinityChainBase = "d5db6f39-fe50-4633-8edc-36e2f3e117e4"
+)
+
+var (
+	DfinityChainId crypto.Hash
+)
+
+func init() {
+	DfinityChainId = crypto.NewHash([]byte(DfinityChainBase))
+}
+
+func VerifyAssetKey(assetKey string) error {
+	if assetKey == DfinityChainBase {
+		return nil
+	}
+	return fmt.Errorf("invalid internet computer asset key %s", assetKey)
+}
+
+func VerifyAddress(addr string) error {
+	if strings.TrimSpace(addr) != addr {
+		return fmt.Errorf("invalid internet computer address %s", addr)
+	}
+
+	buf, err := hex.DecodeString(addr)
+	if err != nil {
+		return err
+	}
```

Comment:
Is there a length requirement for the address?

Reply:
Fixed, the address length is [4]byte + [28]byte